cmd/osbuild-jobsite-builder/main.go

```diff
@@ -166,13 +162,8 @@ func (builder *Builder) HandleProvision(w http.ResponseWriter, r *http.Request)
 		0400,
 	)
 
-	defer func() {
```

Comment:
Why would we remove `defer dst.Close()` here ? It's a stanard pattern and if we don't do it we potentially leak file descriptors  in the error handling of e.g. io.Copy(). Sure it does not matter here because the program dies but I feel it's still worthwhile (code gets refactored etc)

Reply:
It makes the error handling slightly more finicky and less linear to read. Since we always exit we won't be leaking. If this gets refactored into using a deferred close again then yes it can be brought up then.

Reply:
Thanks for the rational. I personally like the go-pattern of "open/check err/defer close" and it's so common that every time I see something that diverges from the pattern my flow is interrupted. Sorry, that's a winded way of saying that I really like consistency in resource/error handling :) (plus if it gets refactored we may not remember about this anymore as it diverges from the usual pattern).